hedera-node/hedera-app/src/main/java/com/hedera/node/app/state/impl/MutableStateBase.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.state.impl;
+
+import com.hedera.node.app.spi.state.WritableState;
+import com.swirlds.common.merkle.MerkleNode;
+import java.util.*;
+import javax.annotation.Nonnull;
+
+/**
+ * A base class for implementations of {@link WritableState}.
+ *
+ * @param <K> The key type
+ * @param <V> The value type
+ */
+abstract class MutableStateBase<K, V> extends StateBase<K, V> implements WritableState<K, V> {
+    /** A map of all modified values buffered in this mutable state */
+    private final Map<K, V> modified = new HashMap<>();
+
+    /** A set of all keys (and their implicit values) removed by this mutable state */
+    private final Set<K> removed = new HashSet<>();
+
+    /**
+     * Create a new StateBase.
+     *
+     * @param stateKey The state key. Cannot be null.
+     */
+    MutableStateBase(@Nonnull final String stateKey) {
```

Comment:
```suggestion
    protected MutableStateBase(@Nonnull final String stateKey) {
```

Reply:
Technically the only subclasses are in the same package, and the class itself is package scoped, so expanding scope from package to `protected` doesn't actually change who can subclass or what they can do. But I do think it could aid in readability to use `protected` (as I do with other methods) to indicate this constructor is meant to be called by subclasses. So I am happy to apply that suggestion. Thanks!